qrenderdoc/Windows/TextureViewer.cpp

```diff
@@ -429,19 +476,79 @@ class TextureListItemModel : public QAbstractItemModel
 
       if(role == Qt::DecorationRole)
       {
-        return QVariant(goArrow);
+        int col = index.column();
+        if(col >= 0 && col < columnCount() && index.row() < rowCount() && ((Column_Count - 1) == col))
+        {
+          return QVariant(goArrow);
+        }
       }
     }
 
     return QVariant();
   }
 
+  template <class T>
+  static bool SortHelper(const T &valA, const T &valB, Qt::SortOrder sortOrder)
+  {
+    return (Qt::SortOrder::DescendingOrder == sortOrder) ? (valA < valB) : !(valA < valB);
+  }
+
+  struct customComparator
+  {
+    static int s_sortCol;
+    static Qt::SortOrder s_sortOrder;
+    static TextureListItemModel *s_tlim;
+
+    inline bool operator()(const TextureDescription &td1, const TextureDescription &td2) const
+    {
+      switch(s_sortCol)
+      {
+        case Column_TexName:
+        {
+          rdcstr name1 = s_tlim->m_Ctx.GetResourceName(td1.resourceId);
+          rdcstr name2 = s_tlim->m_Ctx.GetResourceName(td2.resourceId);
+
+          return SortHelper(name1, name2, s_sortOrder);
+        }
+        case Column_TexWidth: { return SortHelper(td1.width, td2.width, s_sortOrder);
+        }
+        case Column_TexHeight: { return SortHelper(td1.height, td2.height, s_sortOrder);
+        }
+        case Column_TexDepth: { return SortHelper(td1.depth, td2.depth, s_sortOrder);
+        }
+        case Column_MipsCount: { return SortHelper(td1.mips, td2.mips, s_sortOrder);
+        }
+        case Column_TexFormat:
+        {
+          // TODO: Use strings
```

Comment:
What's the reason for this TODO, is there a problem with using strings just now? I agree it's a better way than using the builtin `ResourceFormat` sort (which is not really intended to be user-facing).